stream.go

```diff
@@ -117,9 +135,10 @@ func newClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 		}
 	}
 	callHdr := &transport.CallHdr{
-		Host:   cc.authority,
-		Method: method,
-		Flush:  desc.ServerStreams && desc.ClientStreams,
+		Host:     cc.authority,
+		Method:   method,
+		Flush:    desc.ServerStreams && desc.ClientStreams,
+		FailFast: c.failFast,
```

Comment:
Is it okay to put fail fast into stats.Begin so that we do not need to pass it into the transport layer? You can refactor NewClientStream and newClientStream somehow (e.g., move opt processing to NewClientStream).